### PR TITLE
mouse_m908: init at 3.4

### PR DESCRIPTION
**Pull request by kylelovestoad**
## Description of changes

CLI application to control various Redragon gaming mice

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/linux/mouse_m908/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  '';
+
+  src = fetchFromGitHub {
+    owner = "dokutan";
+    repo = "mouse_m908";
+    rev = "6e48ff76cfe2e199215e4c9a916fbc34619ab227";
+    sha256 = "sha256-sCAvjNpJYkp4G0KkDJtHOBR1vc80DZJtWR2W9gakkzQ=";
+  };
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/os-specific/linux/mouse_m908/default.nix:37]: Better to avoid `with lib`

File: pkgs/os-specific/linux/mouse_m908/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    cp examples/* $out/share/doc/mouse_m908 && \
+    cp README.md $out/share/doc/mouse_m908 && \
+    cp keymap.md $out/share/doc/mouse_m908 && \
+    cp mouse_m908.1 $out/share/man/man1
+  '';
+
+  src = fetchFromGitHub {
+    owner = "dokutan";
+    repo = "mouse_m908";
+    rev = "6e48ff76cfe2e199215e4c9a916fbc34619ab227";
```
Comment (Aleksanaa) [inline pkgs/os-specific/linux/mouse_m908/default.nix:33]: Please reference version (`stdenv.mkDerivation (finalAttrs: {...})`) and `rev = "v${finalAttrs.version}";`

File: pkgs/os-specific/linux/mouse_m908/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    sha256 = "sha256-sCAvjNpJYkp4G0KkDJtHOBR1vc80DZJtWR2W9gakkzQ=";
+  };
+
+  meta = with lib; {
+    homepage = "https://github.com/dokutan/mouse_m908";
+    description = "Control various Redragon gaming mice from Linux, BSD and Haiku";
+    mainProgram = "mouse_m908";
+    license = with licenses; [ gpl3Only ];
```
Comment (Aleksanaa) [inline pkgs/os-specific/linux/mouse_m908/default.nix:41]:
```suggestion
    license = lib.licenses.gpl3Plus;
```

https://github.com/dokutan/mouse_m908/blob/a93f2b81311a5d84ab27f80946cf82eae1e55ef9/CMakeLists.txt#L3

**Comment by gador** [comment]
Result of `nixpkgs-review pr 326060` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mouse_m908</li>
  </ul>
</details>

**Comment by gador** [comment]
##### Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64
- [x] executables tested on x86_64
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] `meta.mainProgram` is set, if applicable.
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

##### Possible improvements

--

##### Comments

LGTM, thanks for your contribution!

File: pkgs/by-name/mo/mouse_m908/package.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    mkdir -p $out/bin \
+      $out/share/doc \
+      $out/share/doc/mouse_m908 \
+      $out/lib/udev/rules.d
+
+    cp mouse_m908 $out/bin/mouse_m908 && \
+    cp mouse_m908.rules $out/lib/udev/rules.d && \
+    cp examples/* $out/share/doc/mouse_m908 && \
+    cp README.md $out/share/doc/mouse_m908 && \
+    cp keymap.md $out/share/doc/mouse_m908 && \
```
Comment (SuperSandro2000) [inline pkgs/by-name/mo/mouse_m908/package.nix:35]: Using && here just makes things complicated.

File: pkgs/by-name/mo/mouse_m908/package.nix [hunk 5]
```diff
@@ -0,0 +1,49 @@
+{ stdenv
+, lib
+, libusb1
+, installShellFiles
+, fetchFromGitHub
+, pkg-config
+}: stdenv.mkDerivation (finalAttrs: {
+  name = "mouse_m908";
```
Comment (SuperSandro2000) [inline pkgs/by-name/mo/mouse_m908/package.nix:8]: This should be pname #330275

**Pull request merged**